crates/sui-framework/packages/move-stdlib/sources/u128.move

```diff
@@ -57,6 +63,31 @@ module std::u128 {
         std::macros::num_sqrt!<u128, u256>(x, 128)
     }
 
+    /// Try to convert a `u128` to a `u8`. Returns `None` if the value is too large.
+    public fun try_as_u8(x: u128): Option<u8> {
+        std::macros::try_as_u8!(x)
+    }
+
+    /// Try to convert a `u128` to a `u16`. Returns `None` if the value is too large.
+    public fun try_as_u16(x: u128): Option<u16> {
+        std::macros::try_as_u16!(x)
+    }
+
+    /// Try to convert a `u128` to a `u32`. Returns `None` if the value is too large.
+    public fun try_as_u32(x: u128): Option<u32> {
+        std::macros::try_as_u32!(x)
+    }
+
+    /// Try to convert a `u128` to a `u64`. Returns `None` if the value is too large.
+    public fun try_as_u64(x: u128): Option<u64> {
+        std::macros::try_as_u64!(x)
+    }
+
+    /// Maximum value for a `u128`
+    public macro fun max_value(): u128 {
```

Comment:
That got me curious! Why are these written as macros?

Reply:
Great question. My hope is to allow them in constants at some point.
We currently don't allow function calls, but we could totally allow macros. And in general since it is a constant, it should always be inlined.

Reply:
ah nice, I get it. thanks!